yamllint/rules/float_values.py

```diff
@@ -0,0 +1,158 @@
+positive and negative infinity and "not a number".
+
+.. rubric:: Options
+
+* Use ``require-numeral-before-decimal`` to require floats to start
+  with a numeral (ex ``0.0`` instead of ``.0``).
+* Use ``forbid-scientific-notation`` to forbid scientific notation.
+* Use ``forbid-nan`` to forbid NaN (not a number) values.
+* Use ``forbid-inf`` to forbid infinite values.
+
+.. rubric:: Default values (when enabled)
+
+.. code-block:: yaml
+
+    rules:
+      float-values:
+        forbid-inf: false
+        forbid-nan: false
+        forbid-scientific-notation: false
+        require-numeral-before-decimal: false
+
+.. rubric:: Examples
+
+#. With ``float-values: {require-numeral-before-decimal: true}``
+
+   the following code snippets would **PASS**:
+   ::
+
+    anemometer:
+      angle: 0.0
+
+   the following code snippets would **FAIL**:
+   ::
+
+    anemometer:
+      angle: .0
+
+#. With ``float-values: {forbid-scientific-notation: true}``
+
+   the following code snippets would **PASS**:
+   ::
+
+    anemometer:
+      angle: 0.00001
+
+   the following code snippets would **FAIL**:
+   ::
+
+    anemometer:
+      angle: 10e-6
+
+#. With ``float-values: {forbid-nan: true}``
+
+   the following code snippets would **FAIL**:
+   ::
+
+    anemometer:
+      angle: .NaN
+
+ #. With ``float-values: {forbid-inf: true}``
+
+   the following code snippets would **FAIL**:
+   ::
+
+    anemometer:
+      angle: .inf
+"""
+
+import re
+
+import yaml
+
+from yamllint.linter import LintProblem
+
+
+ID = 'float-values'
+TYPE = 'token'
+CONF = {
+    'require-numeral-before-decimal': bool,
+    'forbid-scientific-notation': bool,
+    'forbid-nan': bool,
+    'forbid-inf': bool,
+}
+DEFAULT = {
+    'require-numeral-before-decimal': False,
+    'forbid-scientific-notation': False,
+    'forbid-nan': False,
+    'forbid-inf': False,
+}
+
+IS_NUMERAL_BEFORE_DECIMAL_PATTERN = (
+    re.compile('[-+]?(\\.[0-9]+)([eE][-+]?[0-9]+)?')
+)
+IS_SCIENTIFIC_NOTATION_PATTERN = re.compile(
+    '[-+]?(\\.[0-9]+|[0-9]+(\\.[0-9]*)?)([eE][-+]?[0-9]+)'
+)
+IS_INF_PATTERN = re.compile('[-+]?(\\.inf|\\.Inf|\\.INF)')
+IS_NAN_PATTERN = re.compile('\\.nan|\\.NaN|\\.NAN')
+
+
+def check(conf, token, prev, next, nextnext, context):
+    if prev and isinstance(prev, yaml.tokens.TagToken):
+        return
+    if not isinstance(token, yaml.tokens.ScalarToken):
+        return
+    if token.style:
+        return
+    val = token.value
+
+    if conf['forbid-nan'] and IS_NAN_PATTERN.match(val):
+        yield LintProblem(
+            token.start_mark.line + 1,
+            token.start_mark.column + 1,
+            'forbidden not a number value "%s"' % token.value,
+        )
+
+    if conf['forbid-inf'] and IS_INF_PATTERN.match(val):
+        yield LintProblem(
+            token.start_mark.line + 1,
+            token.start_mark.column + 1,
+            f"forbidden infinite value {token.value}",
```

Comment:
This line (and the 2 next messages) should have quotes around the value, like in the first message above.